challenges/2021-08-03-tracing-triangles/TracingTriangles.cs

```diff
@@ -1,11 +1,49 @@
 using Xunit;
+using Xunit.Abstractions;
 
 public class TracingTriangles
 {
+    private readonly ITestOutputHelper output;
+
+    public TracingTriangles(ITestOutputHelper output)
+    {
+        this.output = output;
+    }
+
     public int LongestPath(int[][] triangle)
     {
         // Insert your solution code here
-        return 0;
+        //1. first we have a way to establish what is a child
+        int []max = new int [triangle.Length];
+        output.WriteLine($"Length of Triangle is {triangle.Length}.");
+        output.WriteLine($"Current Length of max: {max.Length}.");
+
+        //2. first element has to have length of 1
+        for (var i = 0; i < triangle.Length; i++)
```

Comment:
Consider: Starting at the top of the triangle and working your way down, you'll have a tough time keeping track of each path weight as you go forward. However, if you work your way *up* the triangle, it's easier to keep track of. Consider a small test example: [[1], [2, 1], [1, 1, 3]], where the largest path weight will be 5, but if you sum too early (from the top), you may assume that the largest path includes the `2` in the second row.